DragonSlayer.cpp

```diff
@@ -1,28 +1,48 @@
 #include "DragonSlayer.h"
 #include "Dragon.h"
+#include "Utility.h"
+#include "AttackItem.h"
+#include <memory>
 
-//DragonSlayer::DragonSlayer
+DragonSlayer::DragonSlayer(std::string name_, int hp_, int armor_) :
+    Character(hp_, armor_, 4),
+    name(name_)
+{
+    helpfulItems = makeHelpfulItems(3);
+    defensiveItems = makeDefensiveItems(1);
+    attackItem = std::make_unique<AttackItem>();
+}
 
-//DragonSlayer::getName
+const std::string& DragonSlayer::getName()
+{
+    return name;
+}
 
 void DragonSlayer::attack(Character& other)
 {
     std::cout << name << " is attacking " << other.getName() << " !!" << std::endl;
     if( auto* dragon = dynamic_cast<Dragon*>(&other) )
     {
-        assert(false);
         //DragonSlayers get a 10x boost when attacking dragons, from their attack item.
         //so they should USE their attack item before attacking the dragon... 
         //note that items are single-use only, so you need to reset it after use.  
         //look in the Character class for how the other item types are reset after use.
         while( dragon->getHP() > 0 )
         {
-          dragon->takeDamage(attackDamage);
+            if( attackItem ) 
+            {
+                useAttackItem( this, attackItem.get() );
+                attackItem.reset();
+            }
+            dragon->takeDamage(attackDamage);
```

Comment:
`USE their attack item before attacking the dragon...`
`dragon->takeDamage(attackDamage)` is attacking the dragon. 
Once you use the attack item, you don't need to check if it's valid anymore.  
Don't check it after every single attack, which is what you're doing in this `while()` loop
use the item BEFORE attacking.
- [ ] revise